### DISCUSSION TITLE
Where can I find the JAR for ESAPI version 1.4 to download?

### DISCUSSION THREAD

**Discussion by leesonwei** [ESAPI/esapi-java-legacy#739] [Q&A] [answered]
Where can I find the JAR for ESAPI version 1.4 to download? Or if you have one, you can send it to me at weilizong2014@hotmail.com. Thank you very much

**Comment by kwwall** [comment, marked as answer]
If it's not in Maven Central, then I'd try the Internet Wayback Machine,
archive.org. Probably need to go back to 2009 or so.

-kevin

**Reply by kwwall** [reply]
@leesonwei - Did this answer your question? If so, please mark it as an answer so everyone will know it's been answered. If it did not answer your question, you should be able to rebuild the 1.4 ESAPI jar from the GitHub source. Unless you have the same Sun JDK version handy--which I think was Java 5 at that time--you are not going to end up generating the same byte code. And even if you did, the bits from the original ESAPI 1.4 jar will differ (because of timestamps in the jar) than whatever you build.

Of course, I would caution you using the ESAPI 1.4 jar for anything production. There were a lot of vulnerabilities (some which have associated CVEs) fixed since that ancient version.
